SyncTests/RecordTests.swift

```diff
@@ -73,6 +78,19 @@ class RecordTests: XCTestCase {
         // Missing ID.
         XCTAssertFalse(Record<CleartextPayloadJSON>.fromEnvelope(EnvelopeJSON(emptyPayload), payloadFactory: clearFactory)!.payload.isValid())
 
+
+        // Non-string ID.
+
+        let invalidEnvelope = EnvelopeJSON(invalidIDRecordString)
+
+        // The envelope is valid...
+        XCTAssertTrue(invalidEnvelope.isValid())
```

Comment:
Fixed. (It was meant to imply "invalidPayloadGUIDEnvelope", but totally did not imply that!)